src/main/java/step2/domain/LottoMachine.java

```diff
@@ -0,0 +1,43 @@
+package step2.domain;
+
+import step2.util.GeneratorUtil;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class LottoMachine {
+    private static final int LOTTO_LAST_NUMBER = 45;
+    private static final int LOTTO_SIZE = 6;
+    private static final int PRICE_PER_LOTTO = 1_000;
+
+    public Lottos createLottos(int inputMoney) {
+        inputCheck(inputMoney);
+        return lottoList(inputMoney/ PRICE_PER_LOTTO);
+    }
+
+    private void inputCheck(int inputMoney) {
+        if (inputMoney < PRICE_PER_LOTTO) {
+            throw new IllegalArgumentException(String.format("로또 구입 금액은 최소 $d 이상이어야 합니다. 입력값 : $d", PRICE_PER_LOTTO, inputMoney));
+        }
+
+        if ((inputMoney % PRICE_PER_LOTTO) != 0) {
+            throw new IllegalArgumentException(String.format("로또 구입 금액은 $d의 배수여야합니다. 입력값 : $d", PRICE_PER_LOTTO, inputMoney));
+        }
+    }
+
+    private Lottos lottoList(int requestCount) {
+        Lottos result = new Lottos();
+        for (int i = 0; i < requestCount; i++) {
+            result.addLotto(generateLotto());
+        }
+        return result;
+    }
```

Comment:
로또 머신에서 생성해서 넣는 방식보다
Lotto의 정적 팩터리 메서드를 사용하여 생성할 수 있으면 좋을 것 같아요

```suggestion
    private Lottos lottoList(int requestCount) {
        Lottos result = new Lottos();
        for (int i = 0; i < requestCount; i++) {
            result.addLotto(Lotto.auto());
        }
        return result;
    }
```

Reply:
이렇게 된다면
로또 일급컬렉션도 아래와 같이 생성할 수 있을 것 같습니다.
`Lottos lottos = Lottos.auto(requestCount);`

Reply:
정적 팩토리 메서드 관련한 피드백은 반영이 안 되어있네요 😔